src/Credentials/ExternalAccountCredentials.php

```diff
@@ -102,6 +103,39 @@ private static function buildCredentialSource(array $jsonKey): ExternalAccountCr
             );
         }
 
+        if (
+            isset($credentialSource['environment_id'])
+            && 1 === preg_match('/^aws(\d+)$/', $credentialSource['environment_id'], $matches)
+        ) {
+            if ($matches[1] !== '1') {
+                throw new InvalidArgumentException(
+                    "aws version \"$matches[1]\" is not supported in the current build."
+                );
+            }
+            if (!array_key_exists('region_url', $credentialSource)) {
+                throw new InvalidArgumentException(
+                    'The region_url field is required for aws1 credential source.'
```

Comment:
region_url should not be required (because we can get it from the env variables in some cases)

Reply:
thank you! This should now be fixed.

Reply:
If `region_url` isn't required, we need to update the AIP: https://google.aip.dev/auth/4117

Are you sure this is the case? I remember reading here (and also in some internal implementation docs) that the `region_url` field is required.

Reply:
Submitted https://github.com/aip-dev/google.aip.dev/pull/1211 to track this